src/hawkmoth/__main__.py

```diff
@@ -45,11 +57,20 @@ def main():
                         choices=['c', 'cpp'],
                         default='c',
                         help='Sphinx domain to be used.')
-    parser.add_argument('--compat',
-                        choices=['none',
-                                 'javadoc-basic',
-                                 'javadoc-liberal',
-                                 'kernel-doc'],
+    compat = parser.add_mutually_exclusive_group()
+    compat.add_argument('--process-docstring',
+                        choices=[
+                            'javadoc',
+                            'napoleon',
+                        ],
+                        help='Process docstring.')
+    compat.add_argument('--compat',
+                        choices=[
+                            'none',
+                            'javadoc-basic',
+                            'javadoc-liberal',
+                            'kernel-doc'
+                        ],
                         help='Compatibility options. See cautodoc_compat.')
```

Comment:
Consider raising `DeprecationWarning` :man_shrugging:

Reply:
Side note: we may also want to institutionalize a deprecation policy so we have a regular, well advertised schedule for dropping things. We have an important user now and everything <.<

Reply:
Yeah, the deprecation has been haphazard so far. If it's release based, I guess it'll have to be major releases. And we're not even at 1.0. :grin: